libs/src/services/creatorNode/index.js

```diff
@@ -417,6 +430,55 @@ class CreatorNode {
     this.authToken = null
   }
 
+  /**
+   * Gets and returns the clock values across the replica set for the wallet in userStateManager.
+   * @returns {Object[]} Array of objects with the structure:
+   *
+   * {
+   *  type: 'primary' or 'secondary',
+   *  endpoint: <Content Node endpoint>,
+   *  clockValue: clock value (should be an integer) or null
+   * }
+   *
+   * 'clockValue' may be null if the request to fetch the clock value fails
+   */
+  async getClockValuesFromReplicaSet () {
+    const user = this.userStateManager.getCurrentUser()
+    if (!user || !user.creator_node_endpoint) {
+      console.error('No user or Content Node endpoint found')
+      return
+    }
+
+    const primary = CreatorNode.getPrimary(user.creator_node_endpoint)
+    const replicaSet = CreatorNode.getEndpoints(user.creator_node_endpoint)
+
+    const clockValueRequest = async (endpoint) => {
+      let type = primary === endpoint ? 'primary' : 'secondary'
+
+      try {
+        const clockValue = await CreatorNode.getClockValue(endpoint, user.wallet)
+        return {
+          type,
+          endpoint,
+          clockValue
+        }
+      } catch (e) {
+        console.error(`Error in getting clock status for ${user.wallet} at ${endpoint}: ${e}`)
+        return {
+          type,
+          endpoint,
+          clockValue: null
+        }
+      }
+    }
+
+    const clockValueResponses = await Promise.all(
+      replicaSet.map(endpoint => clockValueRequest(endpoint))
```

Comment:
can't this just map over returned values from `CreatorNode.getClockValue`?

Reply:
yes! i just thought that wrapping that fn with `clockValueRequest` would prevent any errors from being thrown